docs/developer/index.rst

```diff
@@ -98,6 +98,53 @@ run allows us to generate a coverage report after that::
 The report is in ``report/index.html``, which links to individual
 files and shows exactly which lines were excuted while running the tests.
 
+Run doctests locally
+--------------------
+If `doctestplus <https://pypi.org/project/pytest-doctestplus/>` is installed
+(and it probably is because it's part of
+`sphinx-astropy <https://pypi.org/project/sphinx-astropy/>`,
+which is required to build the documentation locally),
+examples in the documentation are run automatically.
+This serves two purposes:
+
+  - It ensure that the examples we give execute and give the expected output.
+    It is confusing to a user to see an example in the documentation that does not
+    work because it contains bugs to calls functions that have been removed from sherpa.
+  - This also acts as an additional tests.
+
+However, many examples were written because doctestplus was used with Sherpa and do not execute
+as written (e.g. they need implicit set-up that's not spelled out),
+so a number of rst files are explicitly excluded from that run in `pyests.ini`.
+When an rst file is fixed to work top-to-bottom, it should be removed from the
+blacklist in `pytest.ini`.
+
+doctestplus can also run the examples in the doctrings of individual
+functions, but again many files are still blacklisted for this functionality in
+`pytest.ini`.
+During development, you can run doctestplus on individual files like so::
```

Comment:
```suggestion
This serves two purposes:

  - it ensure that the examples we give are actually correct and match the code,
  - and it acts as additional tests of the Sherpa code base.

The `doctest_norecursedirs` setting in the `pytests.ini` file is used to exclude files which can not be
tested. This is generally because the examples were written before doctestplus support was added, and so
they need to be re-worked, or there is too much extra set-up required that would make the examples
hard-to follow. The file should be removed from this list when it has been updated to allow testing with doctestplus.

During development, you can run doctestplus on individual files like so (the option to use depends on whether it is a Python or reStructuredText file)::
```

Is this better? I feel like we don't need to strongly explain why testing is good.